tic_tac_toe/src/board.py

```diff
@@ -0,0 +1,203 @@
+# Board Representation
+
+# test_board = [0, 1, -1, 0, 1, -1, 0, 1, -1]
+# 9 values, for 9 spaces on the board
+#
+# -1 = X
+# 0 = No Value
+# 1 = O
+#
+
+
+class Board:
+    def __init__(self, dim):
+        self.values = [None, None, None, None, None, None, None, None, None]
+
+        self.template = "{}|{}|{}\n-------\n{}|{}|{}\n-------\n{}|{}|{}"
+
+        self.dim = dim
+
+        self.winner = None
+
+    def convert(self, x, y):
+        """
+        converts a 2-dimensional coordinate to a 1-dimensional coordinate
+        Parameters
+        ----------
+        x : integer or float
+        the first of the two 2-dim coordinates to be operated on
+
+        y : integer or float
+        the first of the two 2-dim coordinates to be operated on
+
+        Returns
+        -------
+        integer or float : the final 1-dim coordinate
+        """
+        if x < 0 or y < 0:
+            return False
+
+        elif x > 3 or y > 3:
+            return False
+
+        else:
+            return ((y - 1) * int(self.dim)) + (x - 1)
+
+    def render_board(self):
+        """
+        Renders the board in a way that the user can easily read it
+        Returns
+        -------
+        string : The board rendered with an actual tic tac toe board template
+        """
+
+        # Print out the board, in a way humans can understand
+        return self.template.format(*self.values)
+
+    def validate_move(self, x, y, player):
+        """
+        Takes in a move as 2-dim coordinates, and a player, and validates that it is a valid move
+        Parameters
+        ----------
+        x : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        y : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        player : string
+        "X" or "O", which player
+        Returns
+        ---------
+        : boolean
+        Whether or not it is a valid move
+        """
+        _index = self.convert(x, y)
```

Comment:
Why is this function not just `return self.values[self.convert(x,y)] == None` ?  If you've already moved into that slot in the grid, it's not a valid move, right?

Reply:
👍